### PR TITLE
more inlining of delegates

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

**Comment by DmitryOlshansky** [comment]
Cool stuff.
Still I can't help but wonder - why `const`? Mutable will do just fine b/c in D mutable == thread-local so we can be relatively certain that whatever was initialized recently and never written to can be copy-propagated and inlined. In C++ that wouldn't work because there might be some other references to that object.

**Comment by DmitryOlshansky** [comment]
Also - would it make sense to factor out data-flow primitives or is it too early for that?

**Comment by WalterBright** [comment]
I plan to enhance `onlyOneAssign()` to check for the cases listed in the comment for it. That'll do most of what data flow analysis would. `const` is done first because it doesn't need data flow analysis to prove single assignment.

File: src/expression.d [hunk 1]
```diff
@@ -5848,6 +5848,14 @@ public:
     {
         v.visit(this);
     }
+
+    override void printAST(int indent)
```
Comment (andralex) [inline src/expression.d]: yum, override

File: src/expression.d [hunk 2]
```diff
@@ -8665,6 +8673,14 @@ public:
...
```
Comment (andralex) [inline src/expression.d]: yum

**Comment by andralex** [comment]
Auto-merge toggled on

**Comment by denis-sh** [comment]
Nitpick: Tabs in tests.

**Pull request merged**